prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/git/GitArchive.java

```diff
@@ -0,0 +1,91 @@
+package com.redhat.parodos.tasks.git;
+
+import com.google.common.io.Files;
+import com.redhat.parodos.workflow.parameter.WorkParameter;
+import com.redhat.parodos.workflow.parameter.WorkParameterType;
+import com.redhat.parodos.workflow.task.BaseWorkFlowTask;
+import com.redhat.parodos.workflows.work.DefaultWorkReport;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.eclipse.jgit.api.ArchiveCommand;
+import org.eclipse.jgit.archive.ZipFormat;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.io.FileOutputStream;
+
+@AllArgsConstructor
+public class GitArchive extends BaseWorkFlowTask {
+
+	@Autowired
+	private GitUtils gitUtils;
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		return List.of(WorkParameter.builder().key(this.gitUtils.getGitRepoPath()).type(WorkParameterType.TEXT)
+				.optional(true).description("path where the git repo is located").build());
+	}
+
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		String path = this.gitUtils.getRepoPath(workContext);
+		if (path == null || path == "") {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, new Exception("No valid path: " + path));
+		}
+
+		Repository repo = this.getRepo(path);
+		if (repo == null) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, new Exception("No repository at " + path));
+		}
+
+		try {
+			var archivePath = this.archive(repo);
+			workContext.put(this.gitUtils.getContextArchivePath(), archivePath.toAbsolutePath().toString());
+		}
+		catch (Exception e) {
+			repo.close();
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext,
+					new Exception("Cannot archive the repository:" + e));
+		}
+
+		repo.close();
+		return new DefaultWorkReport(WorkStatus.COMPLETED, workContext, null);
+	}
+
+	private Repository getRepo(String path) {
+		Path gitDir = Paths.get(path);
+		try {
+			Repository repository = new FileRepositoryBuilder().setGitDir(gitDir.toFile()).build();
+			return repository;
+		}
+		catch (Exception e) {
+			return null;
+		}
+	}
+
+	private Path archive(Repository repo) throws Exception {
+
+		// Create a Git instance
+		Git git = new Git(repo);
+		ArchiveCommand.registerFormat("zip", new ZipFormat());
+
+		// Create a ZipFormat instance
+		String tmpdir = Files.createTempDir().getAbsolutePath();
+		Path zipFile = Paths.get(tmpdir + "/output.zip");
+
+		FileOutputStream out = new FileOutputStream(zipFile.toAbsolutePath().toString());
+		git.archive().setTree(repo.resolve("HEAD")).setFormat("zip").setOutputStream(out).call();
+
```

Comment:
1. Should the size be evaluated before?
2. Should `repo.resolve("HEAD")` be parameterized and allow to be set by the dev?
```
SetTree - Set the tag, commit, or tree object to produce an archive for
```

Reply:
at this stage I do not think that it's needed, instead adding more complex can I create a followup issue?

Reply:
sure.